test/helpers/utils.js

```diff
@@ -20,19 +21,43 @@ exports.checkOutput = function (f, argv) {
   console.error = function (msg) { errors.push(msg) }
   console.log = function (msg) { logs.push(msg) }
 
-  var result = f()
-
-  process.exit = _exit
-  process.env = _env
-  process.argv = _argv
-
-  console.error = _error
-  console.log = _log
+  var result
+
+  if (typeof cb === 'function') {
+    process.exit = function () {
+      exit = true
+      cb(null, done())
+    }
+    process.emit = function (ev, value) {
+      if (ev === 'uncaughtException') {
+        done()
+        cb(value)
```

Comment:
Not really, didn't want it to also return the result thought. I don't think it's best practice to include both error and result, and with the move towards promises it won't be possible anymore.